lib/next/validateMessage.js

```diff
@@ -14,24 +14,21 @@ function validateMessage(realMessage, expectedMessage) {
   // for validation with the expected message.
   const real = normalize(coerce(realMessage));
 
-  // Weak coercion applies transformation only to the properties
-  // present in the given message. We don't want to mutate user's assertion.
-  // However, we do want to use the same coercion logic we do
-  // for strict coercion. Thus normalization and coercion are separate.
+  // Use weak coercion on expected message.
+  // This means that only the properties present in expected message
+  // will be coerced. We don't want to mutate user's assertion.
+  // However, we want to use the same coercion logic for any coercion type.
   const expected = normalize(coerceWeak(expectedMessage));
 
-  if (real.statusCode) {
+  if (expected.statusCode) {
     results.statusCode = validateStatusCode(real, expected);
   }
 
-  if (real.headers && expected.headers) {
+  if (expected.headers) {
```

Comment:
I would use `evolve` for `expected` http message, but sometimes the predicate for keys present is different than the actual keys (i.e. it validates `body` when either `body` _or_ `bodySchema` are present). This would complicate the logic so I keep it simple.